packages/vite/src/node/server/hmr.ts

```diff
@@ -169,18 +169,24 @@ export function updateModules(
     ws.send({
       type: 'full-reload'
     })
-  } else {
-    config.logger.info(
-      updates
-        .map(({ path }) => colors.green(`hmr update `) + colors.dim(path))
-        .join('\n'),
-      { clear: true, timestamp: true }
-    )
-    ws.send({
-      type: 'update',
-      updates
-    })
+    return
   }
+
+  if (updates.length === 0) {
+    debugHmr(colors.yellow(`no update happened `) + colors.dim(file))
+    return
+  }
```

Comment:
When `updates` was `[]` the `config.logger.info` below was outputing `22:42:33 [vite] `.